src/Init/Data/Slice/Operations.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+module
+
+prelude
+import Init.Data.Slice.Basic
+import Init.Data.Slice.Notation
+import Init.Data.Iterators
+
+open Std.Iterators
+
+namespace Std.Slice
+
+instance {x : γ} [ToIterator x m β] : ToIterator (Slice.mk x) m β where
+  State := ToIterator.State x m
+  iterMInternal := ToIterator.iterMInternal
+
+/--
+Internal function to obtain an iterator from a slice. Users should import `Std.Data.Iterators`
+and use `Std.Slice.iter` instead.
+-/
+@[always_inline, inline]
+def Internal.iter (s : Slice γ) [ToIterator s Id β] :=
+  ToIterator.iter s
+
+/--
+Returns the number of elements with distinct indices in the given slice.
+
+Example: `#[1, 1, 1][0...2] = 2`.
```

Comment:
What?

Reply:
Oops! Added `.size` and a test that verifies that this actually holds.